File: src/main/java/com/enderzombi102/cmt/CustomMapsTools.java
package com.enderzombi102.cmt;

import com.enderzombi102.cmt.command.OpenGuiCommand;
import com.enderzombi102.cmt.command.ZoneCommand;
import com.enderzombi102.cmt.zone.ZoneManager;
import dev.onyxstudios.cca.api.v3.world.WorldComponentFactoryRegistry;
import dev.onyxstudios.cca.api.v3.world.WorldComponentInitializer;
import net.fabricmc.api.ModInitializer;
import net.fabricmc.fabric.api.client.itemgroup.FabricItemGroupBuilder;
import net.fabricmc.fabric.api.command.v1.CommandRegistrationCallback;
import net.minecraft.item.ItemGroup;
import net.minecraft.item.ItemStack;
import net.minecraft.util.Identifier;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;


public class CustomMapsTools implements ModInitializer, WorldComponentInitializer {

	public static final Logger LOGGER = LogManager.getLogger("CustomMapsTools");
	public static final ItemGroup CMT_GROUP = FabricItemGroupBuilder.build(
			ID("item_group"),
			() -> new ItemStack(CMTContent.invLightItem)
	);

	@Override
	public void onInitialize() {
		LOGGER.info("setting up config page!");
		CMTContent.register();
		CommandRegistrationCallback.EVENT.register( ( dispatcher, dedicated ) -> {
			LOGGER.info("Registering commands!");
			ZoneCommand.register(dispatcher);
			OpenGuiCommand.register(dispatcher);
		});
	}

	@Override
	public void registerWorldComponentFactories(WorldComponentFactoryRegistry registry) {
		registry.register(CMTContent.ZONE_COMP_KEY, ZoneManager::new);
	}

	public static Identifier ID(String path) {
		return new Identifier("cmt", path);
	}
}

File: src/main/kotlin/com/enderzombi102/cmt/CustomMapsTools.kt
package com.enderzombi102.cmt

import com.enderzombi102.cmt.Const.toId
import com.enderzombi102.cmt.command.OpenGuiCommand
import com.enderzombi102.cmt.command.ZoneCommand
import com.enderzombi102.cmt.registry.BlockEntityRegistry
import com.enderzombi102.cmt.registry.BlockRegistry
import com.enderzombi102.cmt.registry.EventListeners
import com.enderzombi102.cmt.registry.ItemRegistry
import net.minecraft.item.ItemGroup
import net.minecraft.item.ItemStack
import org.quiltmc.loader.api.ModContainer
import org.quiltmc.qsl.base.api.entrypoint.ModInitializer
import org.quiltmc.qsl.command.api.CommandRegistrationCallback
import org.quiltmc.qsl.item.group.api.QuiltItemGroup
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class CustomMapsTools : ModInitializer {
	override fun onInitialize(container: ModContainer) {
		LOGGER.info("setting up config page!")
		BlockRegistry.register()
		ItemRegistry.register()
		EventListeners.register()
		BlockEntityRegistry.register()

		CommandRegistrationCallback.EVENT.register { dispatcher, ctx, _ ->
			LOGGER.info("Registering commands!")
			ZoneCommand.register( dispatcher, ctx )
			OpenGuiCommand.register( dispatcher, ctx )
		}
	}

	companion object {
		@JvmField
		val LOGGER: Logger = LoggerFactory.getLogger("CustomMapsTools")
//		val CONFIG = CmtConfig.createAndLoad()
		val ITEM_GROUP: ItemGroup = QuiltItemGroup.builder( "item_group".toId() )
			.icon { ItemStack( ItemRegistry["inv_light"] ) }
			.build()
	}
}
